display_list/BUILD.gn

```diff
@@ -55,51 +55,74 @@ fixtures_location("display_list_benchmarks_fixtures") {
   assets_dir = "$target_gen_dir/"
 }
 
-if (enable_unittests) {
-  executable("display_list_benchmarks") {
-    testonly = true
+source_set("display_list_benchmarks_source") {
+  testonly = true
+
+  sources = [
+    "display_list_benchmarks.cc",
+    "display_list_benchmarks.h",
+  ]
 
-    sources = [
-      "display_list_benchmarks.cc",
-      "display_list_benchmarks.h",
+  deps = [
+    ":display_list",
+    ":display_list_benchmarks_fixtures",
+    "//flutter/benchmarking",
+    "//flutter/common/graphics",
+    "//flutter/fml",
+    "//flutter/testing:skia",
+    "//flutter/testing:testing_lib",
+    "//third_party/dart/runtime:libdart_jit",  # for tracing
+    "//third_party/skia",
+  ]
+
+  defines = []
+
+  if (is_android) {
+    libs = [
+      "android",
+      "EGL",
+      "GLESv2",
+    ]
+  } else {
+    # We only do software benchmarks on non-mobile platforms
+    sources += [
       "display_list_benchmarks_software.cc",
       "display_list_benchmarks_software.h",
     ]
+    defines += [ "ENABLE_SOFTWARE_BENCHMARKS" ]
+  }
 
-    deps = [
-      ":display_list",
-      ":display_list_benchmarks_fixtures",
-      "//flutter/benchmarking",
-      "//flutter/common/graphics",
-      "//flutter/fml",
-      "//flutter/testing:skia",
-      "//flutter/testing:testing_lib",
-      "//third_party/dart/runtime:libdart_jit",  # for tracing
-      "//third_party/skia",
+  # iOS and Fuchsia don't support OpenGL
+  if (!is_fuchsia && !is_ios) {
+    defines += [ "ENABLE_OPENGL_BENCHMARKS" ]
+    sources += [
+      "display_list_benchmarks_gl.cc",
+      "display_list_benchmarks_gl.h",
     ]
+    deps += [ "//flutter/testing:opengl" ]
+  }
+
+  if (is_mac || is_ios) {
```

Comment:
Does this work?

```suggestion
  if ((is_mac && enable_unittests) || is_ios) {
```

Reply:
That would work, but I think it makes more sense to expose the Metal utilities on all platforms. If there isn't a target that depends on it, it won't get built anyway.